Dockerfile

```diff
@@ -1,17 +1,28 @@
-FROM --platform=linux/amd64 debian:buster-slim AS buildbase
+FROM debian:buster-slim AS buildbase
 
-ARG MINICONDA_URL=https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+ARG CONDA_VERSION=latest
 
 WORKDIR /tmp
 
 RUN apt-get update && apt-get install -y wget
 
-RUN wget --quiet $MINICONDA_URL -O ~/miniconda.sh && \
+RUN set -x && \
+    UNAME_M="$(uname -m)" && \
+    if [ "${UNAME_M}" = "x86_64" ]; then \
+        MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-x86_64.sh"; \
+    elif [ "${UNAME_M}" = "s390x" ]; then \
+        MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-s390x.sh"; \
+    elif [ "${UNAME_M}" = "aarch64" ]; then \
+        MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-aarch64.sh"; \
+    elif [ "${UNAME_M}" = "ppc64le" ]; then \
+        MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-ppc64le.sh"; \
+    fi && \
+    wget --quiet $MINICONDA_URL -O ~/miniconda.sh && \
```

Comment:
This inline shell script is worth abstracting into a script in `dev/linux/`, like `bootstrap.sh` or something. Maybe just pass `UNAME_M` and the `/opt/conda` as parameters?

Reply:
Updated the Dockerfile to use built-in args instead of uname and added lots of  comments to make it more clear how the multi-platform stuff works. 

I personally prefer to have the platform/arch related if/else statement within the Dockerfile to have all the important multi-arch related information directly visible on one screen (still less than 100 LOC) without switching files.

Reply:
> Updated the Dockerfile to use built-in args instead of uname and added lots of comments to make it more clear how the multi-platform stuff works.

This is good, serves the same purpose of making it configurable and not a big blob of script in Dockerfile.

> I personally prefer to have the platform/arch related if/else statement within the Dockerfile to have all the important multi-arch related information directly visible on one screen (still less than 100 LOC) without switching files.

Makes sense, thanks for the comments, this works nicely!